internal/pkg/completion/completion.go

```diff
@@ -0,0 +1,119 @@
+package completion
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/spf13/cobra"
+
+	"github.com/devstream-io/devstream/cmd/devstream/list"
+	"github.com/devstream-io/devstream/pkg/util/log"
+)
+
+func ArgsPluginsCompletion(cmd *cobra.Command, args []string, toComplete string) ([]string, cobra.ShellCompDirective) {
+	if len(args) == 0 {
+		return list.PluginsNameSlice(), cobra.ShellCompDirectiveNoFileComp
+	}
+
+	if len(args) == 1 {
+		var choices []string
+		for _, pluginName := range list.PluginsNameSlice() {
+			if strings.HasPrefix(pluginName, args[0]) {
+				choices = append(choices, pluginName)
+			}
+		}
+		return choices, cobra.ShellCompDirectiveNoFileComp
+	}
+
+	return []string{}, cobra.ShellCompDirectiveNoFileComp
+}
+
+func FlagPluginsCompletion(cmd *cobra.Command, flag string) {
+	if err := cmd.RegisterFlagCompletionFunc(flag, func(cmd *cobra.Command, args []string, toComplete string) ([]string, cobra.ShellCompDirective) {
+		return list.PluginsNameSlice(), cobra.ShellCompDirectiveDefault
+	}); err != nil {
+		log.Warn(err)
+	}
+}
+
+func FlagConfigFileCompletion(cmd *cobra.Command) {
```

Comment:
As I said before, it's not mandatory to achieve and you also didn't think it is necessary to achieve, but since you have achieved it, I refer to the official documentation and found that this implementation is not applicable to `Fish` and `PowerShell`

[Fish doesn't support `BashCompFilenameExt` ](https://github.com/spf13/cobra/blob/d8184d32696bee36f682bcb3ace5642af54cd7ad/shell_completions.md#limitations-1)

[Powershell doesn't support `BashCompFilenameExt`](https://github.com/spf13/cobra/blob/d8184d32696bee36f682bcb3ace5642af54cd7ad/shell_completions.md#limitations-2)

A more generic implementation is as follows.

https://github.com/spf13/cobra/blob/d8184d32696bee36f682bcb3ace5642af54cd7ad/shell_completions.md#specify-valid-filename-extensions-for-flags-that-take-a-filename

Of course, if you still don't think it's necessary to implement this feature, you can just remove the relevant code. Thanks again.